File: ML in Python and R/Regression/SLR/SLR.py
import os
os.chdir("/Users/mahaoxi/Desktop/Machine Learning/ML in Python and R/Regression/SLR")

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#Import data
dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,-1].values

#Split into train and test dataset
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y,train_size = 2/3, random_state = 0)

#Fitting SLR to training set
#线性回归package包含了feature scaling
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#Predicting the test response
y_pred = regressor.predict(X_test)

#Visualizing the result
plt.scatter(X_train,y_train,color="r")
plt.plot(X_train,regressor.predict(X_train),color="b")
plt.title("Salary vs Experience (training set)")
plt.xlabel("Years of experience")
plt.ylabel("Salary")
plt.show()#表示图像代码写完，可以显示出来了

plt.scatter(X_test,y_test,color="r")
plt.plot(X_train,regressor.predict(X_train),color="b")
plt.title("Salary vs Experience (test set)")
plt.xlabel("Years of experience")
plt.ylabel("Salary")
plt.show()

regressor.coef_
regressor.intercept_

File: ML in Python and R/Regression/SLR/SLR.R
setwd("/Users/mahaoxi/Desktop/Machine Learning/ML in Python and R/Regression/SLR")

#Import dataset
dataset = read.csv("Salary_Data.csv")

#Split dataset into train and test set
library(caTools)
set.seed(123)
Split<-sample.split(dataset$Salary,SplitRatio = 2/3)
training_set<-subset(dataset,Split == TRUE)
test_set<-subset(dataset,Split ==FALSE)

#Fitting SLR to training set
regressor<-lm(Salary~YearsExperience,data=training_set)
summary(regressor)

#Predict the test 
y_pred<-predict(regressor,test_set)

#Visualizing the training set result
library(ggplot2)
ggplot()+geom_point(aes(x=training_set$YearsExperience,y=training_set$Salary),colour="red")+
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,training_set)),colour="blue",size=1)+
  ggtitle("Salary vs Experience (Training set)")+xlab("Years of experience")+ylab("Salary")

ggplot()+geom_point(aes(x=test_set$YearsExperience,y=test_set$Salary),colour="red")+
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,training_set)),colour="blue",size=1)+
  ggtitle("Salary vs Experience (Test set)")+xlab("Years of experience")+ylab("Salary")
